src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -366,7 +388,22 @@ async Task SearchCoreAsync(Project project)
                 parallel: true,
                 orderedProjects,
                 seenItems,
-                (s, p, cb) => s.SearchCachedDocumentsAsync(p, GetPriorityDocuments(p), _searchPattern, _kinds, _activeDocument, cb, cancellationToken),
+                async (service, projects, onItemFound, onProjectCompleted) =>
+                {
+                    // if the language doesn't support searching cached docs, immediately transition the project to the
+                    // completed state.
+                    if (service is not IAdvancedNavigateToSearchService advancedService)
+                    {
+                        foreach (var project in projects)
+                            await onProjectCompleted().ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        await advancedService.SearchCachedDocumentsAsync(
+                            _solution, projects, GetPriorityDocuments(projects), _searchPattern, _kinds, _activeDocument,
+                            onItemFound, onProjectCompleted, cancellationToken).ConfigureAwait(false);
+                    }
```

Comment:
1. this is what allows F# and TS to no longer have to implement this advanced functinoality
2. if they did, they'd ahve to add those onProjectCompleted loops, which would be pretty ugly.
3. ideally, i can do some more work here and not even add hte progress items for projects that don't support advanced functionality.  but that can be done later.